archunit/src/test/java/com/tngtech/archunit/testutil/assertion/JavaAnnotationAssertion.java

```diff
@@ -58,6 +60,39 @@ public JavaAnnotationAssertion hasAnnotationProperty(String propertyName, Annota
         return this;
     }
 
+    public JavaAnnotationAssertion hasExplicitlyDeclaredStringProperty(String propertyName, String expectedValue) {
+        assertThat(actual.hasExplicitlyDeclaredProperty(propertyName))
+                .as("String @%s.%s()" + context() + " has no explicitly declared value", actual.getRawType().getSimpleName(), propertyName)
+                .isTrue();
+        String actualValue = (String) actual.getExplicitlyDeclaredProperty(propertyName);
+        assertThat(actualValue)
+                .as("String @%s.%s()" + context(), actual.getRawType().getSimpleName(), propertyName)
+                .isEqualTo(expectedValue);
+        assertThat(actual.tryGetExplicitlyDeclaredProperty(propertyName))
+                .as("String @%s.%s()" + context(), actual.getRawType().getSimpleName(), propertyName)
+                .contains(actualValue);
+        return this;
+    }
+
+    public JavaAnnotationAssertion hasNoExplicitlyDeclaredProperty(final String propertyName) {
+        assertThat(actual.hasExplicitlyDeclaredProperty(propertyName))
+                .as("String @%s.%s()" + context() + " has explicitly declared property '%s'", propertyName)
```

Comment:
When this assertion fails, it throws `java.util.MissingFormatArgumentException: Format specifier '%s'`.

I wouldn't worry about a performance impact of eagerly evaluating a few more String concatenations, and would simply always use 
```java
private String annotationPropertyDescription(String typeName, String propertyName) {
    return String.format("%s @%s.%s()", typeName, actual.getRawType().getSimpleName(), propertyName) + context();
}
```
You can find a suggestion on branch `add-API-to-retrieve-explicitly-set-annotation-values.review`.

Reply:
Good point, thanks a lot!

Reply:
(it's a weird mix of concat + format anyway :joy:)

Reply:
(it's a weird mix of concat + format anyway :joy:)